src/Auth/AuthCookie.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Authentication Cookie.
+ *
+ * @package WPGraphQL\Login\Auth
+ * @since 0.0.1
+ */
+
+ // phpcs:disable WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+declare(strict_types=1);
+
+namespace WPGraphQL\Login\Auth;
+
+/**
+ * Class AuthCookie
+ *
+ * Handles custom authentication cookies, including setting the SameSite attribute.
+ */
+class AuthCookie {
+	/**
+	 * Sets the authentication cookies based on user ID.
+	 * Provides an alternative to `wp_set_auth_cookie` that supports the SameSite cookie attribute.
+	 *
+	 * @param int  $user_id  User ID.
+	 * @param bool $remember Whether to remember the user.
+	 */
+	public static function wp_custom_auth_cookie( int $user_id, bool $remember = false ): void {
+		if ( $remember ) {
+			$expiration = time() + apply_filters( 'auth_cookie_expiration', 14 * DAY_IN_SECONDS, $user_id, $remember );
```

Comment:
We should only call these filters once, and docblock them

Reply:
Also may want to break this down via single responsibilities principle, or see if comments are enough to reduce the complexity.

Reply:
Added another private function that gets the cookie_expiration_time in this [commit](https://github.com/AxeWP/wp-graphql-headless-login/pull/129/commits/5eba5efd4d313b5dfb7038c9fbe507345fac92e6)
It's mich better now I think.
Thanks for the feedback